avro/src/main/scala/hydra/avro/registry/RedisSchemaRegistryClient.scala

```diff
@@ -247,7 +305,7 @@ class RedisSchemaRegistryClient(restService: RestService,
     parser.parse(restSchema.getSchemaString)
   }
 
-  override def getBySubjectAndId(s: String, i: Int): Schema = synchronized {
+  override def getBySubjectAndId(s: String, i: Int): Schema = synchronized { //?
```

Comment:
synchronized - it's a potential bottleneck, it's blocking operation. We should avoid it.
Redis has Optimistic locking - we can use schemas versions or...timestamps.
https://redis.io/docs/interact/transactions/#optimistic-locking-using-check-and-set
I'm not sure if Jedis implemented it but we can implement. Otherwise we will have problems and loss of performance.

Reply:
The same approach uses in existing schema registry clients. In case when we have an issue with Redis, it will work like  CachedSchemaRegistryClient and would prefer to use the CachedSchemaRegistryClient approach with synchronisation.